DataDog/dd-trace-py
# feat(config): support libdatadog's library_config for config through file

## Description of changes
- Bumps `libdatadog` to v15.0.0 & handles the required changes to crashtracking
- Starts using the library_config component of `libdatadog` that allows reading configuration from a file **with a higher precedence than environment variables**

### Supported config entries
As of libdatadog v15.0.0 ([source](https://github.com/DataDog/libdatadog/pull/807/files#diff-6011d3279618b70cecf54936d7241711522b09210d2d590a72255357a4698adbR120-R129)):
- `DD_TRACE_DEBUG`
- `DD_SERVICE`
- `DD_ENV`
- `DD_VERSION`
- `DD_PROFILING_ENABLED`

## Testing strategy
Added a unit test.

To test this PR, you can:
1. Create a Linux VM
2. Start this tracer version & ensure everything works well (non-regression)
3. Create a file at the following path: `/etc/datadog-agent/managed/datadog-apm-libraries/stable/libraries_config.yaml`
4. Write the following content in the file:
```yaml
rules:
  - selectors:
      - origin: language
        matches:
          - python
        operator: equals
    configuration:
      DD_TRACE_DEBUG: true
```
5. Start this tracer version
6. Verify that debug logs are properly enabled

## Additional concerns
This change does increase the size of the serverless layer by 380KB (14204KB vs 13824 as the limit) even after optimization. I had to go from `lto=true` to `lto=fast`, as `lto=true` somehow prevents release builds on older machines. That accounts for 200KB, while the 200 other KB are mostly coming from the libdatadog bump.

## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)
